airflow/providers/amazon/aws/hooks/glue_crawler.py

```diff
@@ -0,0 +1,181 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import time
+
+from cached_property import cached_property
+
+from airflow.exceptions import AirflowException
+from airflow.providers.amazon.aws.hooks.base_aws import AwsBaseHook
+
+
+class AwsGlueCrawlerHook(AwsBaseHook):
+    """
+    Interacts with AWS Glue Crawler
+    :param poll_interval: Time (in seconds) to wait between two consecutive calls to check crawler status
+    :type poll_interval: int
+    :param config = Configurations for the AWS Glue crawler
+    :type config = Optional[dict]
+    """
+
+    def __init__(self, poll_interval, *args, **kwargs):
+        self.poll_interval = poll_interval
+        kwargs['client_type'] = 'glue'
+        super().__init__(*args, **kwargs)
+
+    @cached_property
+    def glue_client(self):
+        """:return: AWS Glue client"""
+        return self.get_conn()
+
+    def get_iam_execution_role(self, role_name) -> str:
+        """:return: iam role for crawler execution"""
+        iam_client = self.get_client_type('iam', self.region_name)
+
+        try:
+            glue_execution_role = iam_client.get_role(RoleName=role_name)
+            self.log.info("Iam Role Name: %s", role_name)
+            return glue_execution_role
+        except Exception as general_error:
+            self.log.error("Failed to create aws glue crawler, error: %s", general_error)
+            raise
+
+    def get_or_create_crawler(self, config) -> str:
+        """
+        Creates the crawler if the crawler doesn't exists and returns the crawler name
+
+        :param config = Configurations for the AWS Glue crawler
+        :type config = Optional[dict]
+        :return: Name of the crawler
+        """
+        self.get_iam_execution_role(config["Role"])
+
+        try:
+            self.glue_client.get_crawler(**config)
+            self.log.info("Crawler already exists")
+            try:
+                self.glue_client.update_crawler(**config)
+                return config["Name"]
+            except Exception as general_error:
+                self.log.error("Failed to update aws glue crawler, error: %s", general_error)
+                raise
+        except self.glue_client.exceptions.EntityNotFoundException:
+            self.log.info("Creating AWS Glue crawler")
+            try:
+                self.glue_client.create_crawler(**config)
+                return config["Name"]
+            except Exception as general_error:
+                self.log.error("Failed to create aws glue crawler, error: %s", general_error)
+                raise
```

Comment:
The nested trys get a bit hard to decipher.  

```suggestion
        crawler_name = config["Name"]
        try:
            self.glue_client.get_crawler(Name=crawler_name)
            self.log.info(f"Crawler '{crawler_name}' already exists; updating crawler.")
            self.glue_client.update_crawler(**config)
        except self.glue_client.exceptions.EntityNotFoundException:
            self.log.info("Creating AWS Glue crawler")
            self.glue_client.create_crawler(**config)
        return crawler_name
```

I think it's better to remove the try except when you are just catching and immediately raising because it impairs readability without benefit --- it will be clear enough in the logs that the operation failed.